### PR TITLE
Cyber Connections Operator (NINJA REWRITE) [DNM]

### PR DESCRIPTION
**Pull request by kilkun**
-Refactored parts of old Ninja code to actually work and not be 100%
certifiable shit
-Renamed Ninja to "Cyber Connections Operator"
-Rewrote Ninja Lore.
-Collected all Ninja code into a single massive .dm
-All of Ninja_Equipment.dm poured over and hand translated.
-AI functions removed from suit to prevent feature bloat.
-Cell draining is now no longer a special snowflake check added directly
into cell code, is instead a 2nd step of an If-Else statement and is the
same process as upgrading a suit cell

Lookan for feedback. Am I doing a decent job so far? Pls no bully.

### CODE REVIEW

File: code/modules/operator/main.dm [hunk 1]
```diff
@@ -0,0 +1,1024 @@
...
+
+/*---------------------------------------------------------------------------------------
+
+//////////////////        VERBS!          ///////////////////////////////////////////////
+
+---------------------------------------------------------------------------------------*/
+
+/obj/item/clothing/suit/space/operator/New()//for when a new suit is made!
+	..()
+	verbs += /obj/item/clothing/suit/space/operator/proc/init //suit init verb
```
Comment (clusterfack) [inline code/modules/operator/main.dm]: verbs are a thing

File: code/modules/operator/main.dm [hunk 2]
```diff
@@ -0,0 +1,1024 @@
...
+	cell = /obj/item/weapon/cell/hyper
+	cell.charge = 15000
+
+/obj/item/clothing/suit/space/operator/proc/terminate() //this deletes stuff
+	qdel(o_hood)
+	del(o_gloves)
```
Comment (clusterfack) [inline code/modules/operator/main.dm]: delete sucks

**Comment by clusterfack** [comment]
Ugh this is too much just kill me

**Comment by kilkun** [comment]
I'm sorry blame the tg nerd who wrote this shit

File: code/modules/operator/main.dm (hunk 1 above)
Comment (kilkun) [inline code/modules/operator/main.dm]: giving a verb that calls a proc that does nothing but call another proc.

it's called ironic coding. GOD. *vaping intensifies*

File: code/modules/operator/main.dm (hunk 2 above)
Comment (kilkun) [inline code/modules/operator/main.dm]: literally used qdel in the line above, only to switch back to del.

I have no words, only shame.

**Comment by RemieRichards** [comment]
Don't -entirely- blame me, for example the qdel and then del thing a line below is unique to you guys.

**Comment by Ephx1** [comment]
-Rewrote Ninja Lore.
But why?

-Renamed Ninja to "Cyber Connections Operator"
BUT WHY?

**Comment by kilkun** [comment]
Because remove weeaboo.

**Comment by Ephx1** [comment]
Cyber Connections Operator sounds like a guy who sits behind a desk to tell people to restart their computer every time they call in.

**Comment by kilkun** [comment]
Space Ninja sounds like a dude in a spacesuit trying to sneak through space before getting instantly found and shot to death :P

**Comment by clusterfack** [comment]
I know we talked about lots of the things that need to be fixed on IRC but I missed mentioning quick deletion, any references to another object and to the object you have should be removed in a Destroy() proc to make qdel() work.

**Comment by BestName** [comment]
>Space Ninja sounds like a dude in a spacesuit trying to sneak through space before getting instantly found and shot to death :P 

Well the name fits the role

**Comment by DeityLink** [comment]
Rename it back to Space Ninja pls.

**Comment by Hoshi-chan** [comment]
"Cyber Connections Operator" is a retarded name.

**Comment by kilkun** [comment]
no space ninja is 1000x worse and the lore regarding nanotrasen-sized
megacorps needs expanding.

**Comment by Zth--** [comment]
Space Ninja is 1000x better

**Comment by kilkun** [comment]
If u want dat shitty name u can code it urself

**Comment by Iamgoofball** [comment]
>doesn't completely delete ninja

no

**Comment by Iamgoofball** [comment]
ps. the name sounds like he's a fucking fedora tipping neckbeard

**Comment by BestName** [comment]
How do you delete something that's already been deleted!?!?!?!?!?!???
#5228

**Comment by Iamgoofball** [comment]
I don't see no head wound judging by this pr

**Comment by kilkun** [comment]
And space ninja doesnt lmao?

**Comment by Kurfursten** [comment]
This can probably safely be closed, as kilkun has been perma'd and probably(?) isn't going to continue development for a server he can't play on.

**Comment by kilkun** [comment]
Nah, development will continue soonish.

I put too much work into this to just drop it.

**Comment by ihadtoregisterforthis** [comment]
Only if you don't tell him about the unbanning script included.

Also, you keep referring to the call center operator as ninja, but don't want it to be one.

This will be called ninja by everyone BTW tbh fam Holmes IMHO smh mate ayy roflmao etc.

**Comment by kilkun** [comment]
>Also, you keep referring to the call center operator as ninja, but don't want it to be one.

Context is everything.

>This will be called ninja by everyone

Well I'm banned so that doesn't really matter to me.

I'm sorry but the ineherent lack of megacorp lore expansion is pretty atrocious.

I might at some point when nearing future completion, key emphasis on MIGHT, consider "Cyborg Ninja" as a slang name, seeing as 99% of the inspiration behind the antag is Grey Fox.

**Comment by Hoshi-chan** [comment]
I'd be okay with "Cyborg Ninja" because that's what I've always called it the thousands of times that I've prayed for it.

**Comment by gbasood** [comment]
the only old PR without conflicts
wew

**Comment by DeityLink** [comment]
That's because we've beforehand removed (almost) every single reference to space ninjas from the code.

I'll reiterate though that unless the role is called Space Ninja or Cyborg Ninja I'm against it.

**Comment by ihadtoregisterforthis** [comment]
tumblr_mi4kypxrfy1s01owao1_1280

**Comment by DeityLink** [comment]
nin

**Comment by PJB3005** [comment]
Comment if you want it reopened.

**Pull request closed without merging**